Libraries/Lists/VirtualizedSectionList.js

```diff
@@ -20,34 +20,24 @@ import type {Props as VirtualizedListProps} from 'VirtualizedList';
 
 type Item = any;
 type SectionItem = any;
+type Option = any;
 
-type SectionBase = {
-  // Must be provided directly on each section.
-  data: $ReadOnlyArray<SectionItem>,
-  key?: string,
-
-  // Optional props will override list-wide props just for this section.
-  renderItem?: ?({
-    item: SectionItem,
-    index: number,
-    section: SectionBase,
-    separators: {
-      highlight: () => void,
-      unhighlight: () => void,
-      updateProps: (select: 'leading' | 'trailing', newProps: Object) => void,
-    },
-  }) => ?React.Element<any>,
-  ItemSeparatorComponent?: ?React.ComponentType<any>,
-  keyExtractor?: (item: SectionItem, index: ?number) => string,
-
-  // TODO: support more optional/override props
-  // FooterComponent?: ?ReactClass<any>,
-  // HeaderComponent?: ?ReactClass<any>,
-  // onViewableItemsChanged?: ({viewableItems: Array<ViewToken>, changed: Array<ViewToken>}) => void,
-};
+type SectionBase = any;
```

Comment:
These `any`s are unfortunate - how tricky is it to use more specific types?

Reply:
Unfortunately, I don't think we can specify the type of SectionBase, because sections can be plain JS, Immutable or any data blob, so, by exemple the object tree of Immutable is really different of plain JS. If we keep this definition, and the user specify something else than plain JS, a lot of warning would appear. I understand, that keeping this kind of type definition is important, but in this case I don't find any other solutions, maybe I'm wrong, but if you know a way to handle it, let me know, and I will try to implement it.

Reply:
There isn't anything more specific we can assume about it? Isn't it at least always an Object?

Reply:
You're right, it can be an object! :) I had misread the declaration of `sections` type and confused with the flow type of an element in the array.
I will change it! Sorry